modules/ui/src/app/components/download-report-zip/download-report-zip.component.spec.ts

```diff
@@ -13,75 +13,153 @@
+            profiles: [],
+          },
+          autoFocus: true,
+          hasBackdrop: true,
+          disableClose: true,
+          panelClass: 'initiate-test-run-dialog',
+        });
+
+        tick();
+
+        expect(testrunServiceMock.downloadZip).toHaveBeenCalled();
+        openSpy.calls.reset();
+      }));
+
+      it('should navigate to risk profiles page if profile is null', fakeAsync(() => {
+        const openSpy = spyOn(component.dialog, 'open').and.returnValue({
+          afterClosed: () => of(null),
+        } as MatDialogRef<typeof DownloadZipModalComponent>);
+
+        fixture.ngZone?.run(() => {
+          component.onClick();
+
+          expect(openSpy).toHaveBeenCalledWith(DownloadZipModalComponent, {
+            ariaLabel: 'Download zip',
+            data: {
+              hasProfiles: false,
+              profiles: [],
+            },
+            autoFocus: true,
+            hasBackdrop: true,
+            disableClose: true,
+            panelClass: 'initiate-test-run-dialog',
+          });
+
+          tick();
+
+          expect(router.url).toBe(Routes.RiskAssessment);
+          openSpy.calls.reset();
+        });
+      }));
+
+      it('should do nothing if profile is undefined', fakeAsync(() => {
+        const openSpy = spyOn(component.dialog, 'open').and.returnValue({
+          afterClosed: () => of(undefined),
+        } as MatDialogRef<typeof DownloadZipModalComponent>);
+
+        component.onClick();
+
+        expect(openSpy).toHaveBeenCalledWith(DownloadZipModalComponent, {
+          ariaLabel: 'Download zip',
+          data: {
+            hasProfiles: false,
+            profiles: [],
+          },
+          autoFocus: true,
+          hasBackdrop: true,
+          disableClose: true,
+          panelClass: 'initiate-test-run-dialog',
+        });
+
+        tick();
+
+        expect(testrunServiceMock.downloadZip).not.toHaveBeenCalled();
+        openSpy.calls.reset();
+      }));
     });
+  });
 
-    describe('with data provided', () => {
-      beforeEach(() => {
-        component.data = MOCK_PROGRESS_DATA_COMPLIANT;
-        fixture.detectChanges();
-      });
+  describe('DOM tests', () => {
+    it('should open risk profiles modal on click', () => {
+      const openSpy = spyOn(component.dialog, 'open');
+      compiled.click();
 
-      it('should have download report link', () => {
-        const downloadReportLink = compiled.querySelector(
-          '.download-report-link'
-        ) as HTMLAnchorElement;
-
-        expect(downloadReportLink).not.toBeNull();
-        expect(downloadReportLink.href).toEqual(
-          'https://api.testrun.io/export.pdf'
-        );
-        expect(downloadReportLink.download).toEqual(
-          'delta_03-din-cpu_1.2.2_compliant_22_jun_2023_9:20'
-        );
-      });
+      expect(openSpy).toHaveBeenCalled();
+
+      openSpy.calls.reset();
     });
   });
 });
+
+@Component({
+  selector: 'app-fake-risk-assessment-component',
```

Comment:
Could you explain why do we need it here?

Reply:
Navigator can't navigate without providing component in router